src/core/dataflow/src/DataFlowEngine.cpp

```diff
@@ -380,23 +377,29 @@ DataFlowResult *DataFlowEngine::applyCustomizableForwardAnalysis(
     /*
      * Compute OUT[inst]
      */
-    auto oldSize = outSetOfInst.size();
+    auto oldSizeOut = outSetOfInst.size();
     computeOUT(inst, outSetOfInst, df);
 
     /* Check if the OUT of the first instruction of the current basic block
      * changed.
      */
-    if (false || (!alreadyVisited[inst]) || (outSetOfInst.size() != oldSize)) {
-      alreadyVisited[inst] = true;
+    if ((computedOnce.find(bb) == computedOnce.end())
```

Comment:
Why is this correct? 
A basic block might need to be re-evaluated multiple times.

Reply:
Yes, 
The if condition here shown by git diff is not complete, the complete condition should be
` if ((computedOnce.find(bb) == computedOnce.end())
        || (outSetOfInst.size() != oldSizeOut)) `

If the basic block is not computed before, or any changes (out set in forward) happen in the basic block, that bb will go into the recomputing. Otherwise, bb is computed and nothing change, it will skip.